src-v2/Judge/NTS.Judge.Blazor/Pages/Dashboard/Startlist/StartlistTimer.razor

```diff
@@ -22,33 +23,47 @@
 
     protected override void OnInitialized()
     {
-        _timer = new Timer(TimerRefreshInMilliseconds);
+        _timer = new Timer(TimerRefreshTime);
+        _timer.Elapsed += OnTimerElapsed;
+        _timer.AutoReset = (_time > TimeSpan.FromMinutes(StopTimerAtTime));
     }
 
     protected override void OnParametersSet()
     {
-        _time = TimeSpan.Parse(StartTime);
+        var now = DateTime.Now.TimeOfDay;
+        _time = StartTime - now;
         FormatTime();
-        _timer.Elapsed += OnTimerElapsed;
-        _timer.AutoReset = (_time > TimeSpan.FromMinutes(StopTimerAtTime));
         _timer.Enabled = _time > TimeSpan.FromMinutes(StopTimerAtTime);
     }
 
     private async void OnTimerElapsed(object? sender, System.Timers.ElapsedEventArgs e)
     {
-        _time = _time.Subtract(TimeSpan.FromSeconds(1));
+        _time = _time.Subtract(TimerRefreshTime);
         FormatTime();
         if (_time <= TimeSpan.FromMinutes(StopTimerAtTime))
         {
             _timer.Enabled = false;
         }
-        _displayTime = _time.ToString().Replace("-", " - ");
         await Render();
     }
 
-    public void FormatTime()
-    {
-        _displayTime = _time.ToString().Replace("-", " - ");
+    private void FormatTime()
+    {        
+        var day = DateOnly.FromDateTime(DateTime.Now);
+        if (_time >= TimeSpan.Zero)
+        {
+            var time = TimeOnly.FromTimeSpan(_time);
+            _displayTime = new DateTime(day, time).ToString("HH:mm:ss");
+        }
+        else
+        {
+            var positiveSignedTime = _time.Negate();
+            var time = TimeOnly.FromTimeSpan(positiveSignedTime);
+            _displayTime = new DateTime(day, time).ToString("HH:mm:ss");
+            if(_time <= TimeSpan.FromSeconds(-1)){
+                _displayTime = " - " + _displayTime;
+            }
+        }
```

Comment:
This logic seems very obfuscated, probably because you are using a `TimeSpan` for the `StartTime` value. Also it's not formatter very well.